packages/package-requester/src/packageRequester.ts

```diff
@@ -492,6 +493,17 @@ function fetchToStore (
   }
 }
 
+async function writeJsonFile (filePath: string, data: Object) {
+  const targetDir = path.dirname(filePath)
+  // TODO: use the API of @pnpm/cafs to write this file
+  // There is actually no need to create the directory in 99% of cases.
+  // So by using cafs API, we'll improve performance.
```

Comment:
I have not done it in this PR because it needs some significant refactoring. This fix needs to be shipped fast.